flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/NoFetchingInput.java

```diff
@@ -73,17 +73,14 @@ protected int require(int required) throws KryoException {
         position = 0;
         int bytesRead = 0;
         int count;
-        while (true) {
+        while (bytesRead < required) {
             count = fill(buffer, bytesRead, required - bytesRead);
```

Comment:
@dannycranmer I saw this linked to in slack - I wondered whether we could have junits for the required case and higher than required? Maybe as a follow on issue / pr ? I could code if you were willing to merge for me?